src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -378,7 +362,7 @@ internal sealed class AsyncForwardEntryPoint : SynthesizedEntryPointSymbol
 
             public override TypeSymbol ReturnType => _getAwaiterGetResultCall.Type;
 
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
```

Comment:
Seems like we have a bit of a difference now.  The `AsyncForwardEntryPoint` makes the `binder.GetAwaitableExpressionInfo` call in the ctor, while `ScriptEntryPoint` does it inside of `CreateBody`.  At a glance I would expect both types to do this the same way.  Is there a reason for this difference?